### DISCUSSION TITLE
Track Cell changes

### DISCUSSION THREAD

**Discussion by hussamkhatib** [nick-keller/react-datasheet-grid#279] [Feature requests]
Feature request:
Ability to track cell changes in a similar way [rows are tracked](https://react-datasheet-grid.netlify.app/docs/examples/tracking-rows-changes).
 
 The Problem: 
 The `onChange` handler callback is not usefull here since it only has the newValue and indexes, which is not enough to tell which cell changed. 

solution:
return the columns that are being updated as part of the onChange handler.

**Comment by nick-keller** [comment]
You can listen to the [onFocus](https://react-datasheet-grid.netlify.app/docs/api-reference/props#onfocus) event and store which cell is beeing edited.
But since cells can depend on one another, I would suggest to diff the rowws to know which ones have changed insead of relying on the cells that is being edited. As a benefit, if a cell is edited back to its previous value this would consider it as "un-changed" ;)
